server/data/data-library/src/main/java/org/apache/james/repository/file/AbstractFileRepository.java

```diff
@@ -188,7 +188,7 @@ public Repository getChildRepository(String childName) {
             throw new RuntimeException("Cannot initialize child " + "repository " + childName + " : " + e);
         }
 
-        if (DEBUG) {
+        if (DEBUG && LOGGER.isDebugEnabled()) {
```

Comment:
why ``if (DEBUG)`` is not enough ?

Reply:
I agree with you, the aim of the _DEBUG_ variable is not clear.
BTW, depending on what you want to print, it's a good habit to have a condition before calling `LOGGER.debug`
No in this case, but in some cases, formatting the string to be logged is time consuming (extracting things from stream...).

To summarize:
- in this case I agree that we don't need the condition
- in some cases, a condition on the logger level is needed